To: vim_dev@googlegroups.com
Subject: Patch 7.3.304
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.304
Problem:    Strawberry Perl doesn't work on MS-Windows.
Solution:   Use xsubpp if needed. (Yasuhiro Matsumoto)
Files:	    src/Make_ming.mak, src/Make_mvc.mak


*** ../vim-7.3.303/src/Make_ming.mak	2011-06-19 01:30:01.000000000 +0200
--- src/Make_ming.mak	2011-09-14 10:44:55.000000000 +0200
***************
*** 108,113 ****
--- 108,120 ----
  # on NT, it's here:
  PERLLIB=$(PERL)/lib
  PERLLIBS=$(PERLLIB)/Core
+ XSUBPP=$(PERLLIB)/ExtUtils/xsubpp
+ XSUBPP_EXISTS=$(shell perl -e "print 1 unless -e '$(XSUBPP)'")
+ ifeq "$(XSUBPP_EXISTS)" ""
+ XSUBPP=perl $(XSUBPP)
+ else
+ XSUBPP=xsubpp
+ endif
  endif
  
  # uncomment 'LUA' if you want a Lua-enabled version
***************
*** 696,702 ****
  endif
  
  if_perl.c: if_perl.xs typemap
! 	perl $(PERLLIB)/ExtUtils/xsubpp -prototypes -typemap \
  	     $(PERLLIB)/ExtUtils/typemap if_perl.xs > $@
  
  $(OUTDIR)/netbeans.o:	netbeans.c $(INCL) $(NBDEBUG_INCL) $(NBDEBUG_SRC)
--- 703,709 ----
  endif
  
  if_perl.c: if_perl.xs typemap
! 	$(XSUBPP) -prototypes -typemap \
  	     $(PERLLIB)/ExtUtils/typemap if_perl.xs > $@
  
  $(OUTDIR)/netbeans.o:	netbeans.c $(INCL) $(NBDEBUG_INCL) $(NBDEBUG_SRC)
*** ../vim-7.3.303/src/Make_mvc.mak	2011-06-12 20:31:25.000000000 +0200
--- src/Make_mvc.mak	2011-09-14 10:46:08.000000000 +0200
***************
*** 785,790 ****
--- 785,795 ----
  PERL_INC = /I $(PERL_INCDIR)
  PERL_OBJ = $(OUTDIR)\if_perl.obj $(OUTDIR)\if_perlsfio.obj
  XSUBPP = $(PERL)\lib\ExtUtils\xsubpp
+ !if exist($(XSUBPP)
+ XSUBPP = $(PERL_EXE) $(XSUBPP)
+ !else
+ XSUBPP = xsubpp
+ !endif
  XSUBPP_TYPEMAP = $(PERL)\lib\ExtUtils\typemap
  
  !endif
***************
*** 1041,1047 ****
  	$(CC) $(CFLAGS) $(LUA_INC) if_lua.c
  
  if_perl.c : if_perl.xs typemap
! 	$(PERL_EXE) $(XSUBPP) -prototypes -typemap $(XSUBPP_TYPEMAP) \
  		-typemap typemap if_perl.xs > if_perl.c
  
  $(OUTDIR)/if_perl.obj: $(OUTDIR) if_perl.c  $(INCL)
--- 1046,1052 ----
  	$(CC) $(CFLAGS) $(LUA_INC) if_lua.c
  
  if_perl.c : if_perl.xs typemap
! 	$(XSUBPP) -prototypes -typemap $(XSUBPP_TYPEMAP) \
  		-typemap typemap if_perl.xs > if_perl.c
  
  $(OUTDIR)/if_perl.obj: $(OUTDIR) if_perl.c  $(INCL)
*** ../vim-7.3.303/src/version.c	2011-09-08 23:24:09.000000000 +0200
--- src/version.c	2011-09-14 10:47:41.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     304,
  /**/

-- 
It's totally unfair to suggest - as many have - that engineers are socially
inept.  Engineers simply have different objectives when it comes to social
interaction.
				(Scott Adams - The Dilbert principle)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
